Ensure the right context is active on map cleanup

ivovandongen opened the pull request:
Fixes #9572

Could alternatively be done in the renderer frontend, but that would require passing the backend to it just for cleanup.

jfirebaugh commented:
Seems like a better fix would be to change`Renderer::Impl::~Impl()`, removing `backend.getScopeType()` and using the default `BackendScope::ScopeType::Explicit`.

Same with `Renderer::Impl::onLowMemory()` -- that needs to be `BackendScope::ScopeType::Explicit`.

ivovandongen commented:
@jfirebaugh Thanks, you're right. That is much cleaner.

I've:
- Removed the previous changes
- Fixed the scope type in `Renderer::Impl::~Impl()`
- Fixed the scope type in `Renderer::Impl::onLowMemory`
- Replaced the scope handling in `Renderer::Impl::render` with an assertion
- Removed `RendererBackend::getScopeType()`
- Updated all platforms to handle the `BackendScope` in the `RendererFrontend` implementation on render

I didn't add a `BackendScope&` parameter to `Renderer::render` as we don't actually use it in the renderer directly, but added a assertion instead like mentioned above.
